modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/FilesystemFriendlyNameGenerator.java

```diff
@@ -0,0 +1,23 @@
+package org.testcontainers.junit.jupiter;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import static org.junit.platform.commons.util.StringUtils.isBlank;
+
+class FilesystemFriendlyNameGenerator {
+    private static final String UNKNOWN_NAME = "unknown";
+    private static final String ALLOWED_CHARACTERS_REGEX = "[^\\w -]";
+
+    static String filesystemFriendlyNameOf(ExtensionContext context) {
+        String displayName = context.getDisplayName();
```

Comment:
what about `getUniqueId()` with some escaping?

Reply:
Display name seems to be too risky and not unique enough. Plus, the current approach (drop non-alpha characters) can be problematic and lead to filename conflicts

Reply:
@bsideup I like your suggestion and updated the PR.